### PR TITLE
WIP Refactor streaming client

### PR DESCRIPTION
**Pull request by berzoidberg**
The streaming client library is huge and a fair bit unwieldy. I'm working on a complete refactor that will reduce the code by about 80%. This may introduce some backwards-incompatible changes.

### CODE REVIEW

File: examples/streaming/timesales.py [hunk 1]
```diff
@@ -38,17 +66,26 @@ def initialize(self):
         """
         self.tda_client = easy_client(
             api_key=self.api_key,
-            redirect_uri='https://localhost:8080',
-            token_path=self.credentials_path)
-        self.stream_client = StreamClient(self.tda_client, account_id=self.account_id)
+            redirect_uri="https://localhost:8080",
+            token_path=self.credentials_path,
```
Comment (hmanfarmer) [inline examples/streaming/timesales.py:70]: Is that an errant extra comma at the end on line 70?
`token_path=self.credentials_path,`

**Comment by hmanfarmer** [comment]
10 of the 48 files touched in this PR don't have much to do with the streaming client refactor. Specifically:

- Birthday_Dividends example
- Logging changes
- Import statement adjustments (this includes all of the orders files that were touched)

Maybe it is a bit OCD of me, but me thinks those 10 files should be put into a seperate PR, mainly to get rid of the noise in this PR since there is already a lot going on.

File: examples/streaming/timesales.py (hunk 1 above)
Comment (hmanfarmer) [reply examples/streaming/timesales.py:70]: I just now learned that trailing commas are allowed in python. Ignore my comment.

**Comment by berzoidberg** [comment]
> 10 of the 48 files touched in this PR don't have much to do with the streaming client refactor. Specifically:
> 
>     * Birthday_Dividends example
> 
>     * Logging changes
> 
>     * Import statement adjustments (this includes all of the orders files that were touched)
> 
> 
> Maybe it is a bit OCD of me, but me thinks those 10 files should be put into a seperate PR, mainly to get rid of the noise in this PR since there is already a lot going on.

This branch is based off of another branch that will be merged first, so many of those changes will be upstreamed first. I'm doing a lot in rapid succession so things look a little weird.

**Comment by r4m0n** [comment]
Is there any interest in continuing this PR? I need to use some other APIs that aren't currently implemented in the streaming client and stumbled in some bad architecture issues in it trying to add the new commands...

This PR structure is a lot clearer (and also contains what I need already), so I'm up for taking it over, cleaning it up and making a new PR, unless someone is willing to finish this one up.

**Pull request closed without merging**